src/config_file.c

```diff
@@ -1423,7 +1423,7 @@ static int config_write(diskfile_backend *cfg, const char *key, const regex_t *p
 	while (!reader->eof) {
 		c = reader_peek(reader, SKIP_WHITESPACE);
 
-		if (c == '\0') { /* We've arrived at the end of the file */
+		if (c == '\n') { /* We've arrived at the end of the file */
```

Comment:
This is a subtle thing - `reader_peek` returns `\n` on EOF, and when using `SKIP_WHITESPACE`, *only* on EOL.  I presume that this once returned `\0` and due to some funny interactions with `getchar` it changed.  I added the test case to ensure that we handle EOL correctly to avoid regressions...